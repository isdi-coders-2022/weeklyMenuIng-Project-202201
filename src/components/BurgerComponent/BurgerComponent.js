import styled from "styled-components";

const burgerHeight = 33;

const BurgerContainer = styled.div`
  display: flex;
  flex-direction: column;
  height: ${burgerHeight}px;
  width: min-content;
  justify-content: space-between;
  &:hover .burger-bar:nth-child(2),
  &.active .burger-bar:nth-child(2) {
    opacity: 0;
  }
  &:hover .burger-bar:nth-child(1),
  &.active .burger-bar:nth-child(1) {
    transform: rotate(45deg) translate(7.5px, ${burgerHeight / 2 - 4.5}px);
  }
  &:hover .burger-bar:nth-child(3),
  &.active .burger-bar:nth-child(3) {
    transform: rotate(-45deg) translate(7.5px, -${burgerHeight / 2 - 4.5}px);
  }
`;

const BurgerBar = styled.span`
  position: relative;
  width: 35px;
  background-color: #0f0f0f;
  height: 5px;
  border-radius: 5px;
  transition: all 0.2s;
`;

const BurgerComponent = ({ isActive, actionOnClick }) => {
  return (
    <BurgerContainer
      title="Navigation Menu"
      onClick={actionOnClick}
      className={`burger-container${isActive ? " active" : ""}`}
    >
      <BurgerBar className="burger-bar" />
      <BurgerBar className="burger-bar" />
      <BurgerBar className="burger-bar" />
    </BurgerContainer>
  );
};

export default BurgerComponent;
