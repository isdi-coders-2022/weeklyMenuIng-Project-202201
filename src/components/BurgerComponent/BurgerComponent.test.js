import { render, screen } from "@testing-library/react";
import BurgerComponent from "./BurgerComponent";
import userEvents from "@testing-library/user-event";
import TestRenderer from "react-test-renderer";

describe("Given a BurgerComponent component", () => {
  describe("When it's rendered passing a function", () => {
    test("Then the action should be called on click", () => {
      const action = jest.fn();
      const titleText = "Navigation Menu";

      render(<BurgerComponent actionOnClick={action} />);
      const burger = screen.queryByTitle(titleText);
      userEvents.click(burger);

      expect(action).toHaveBeenCalled();
    });

    test("then it should match the snapshot", () => {
      const app = TestRenderer.create(<BurgerComponent />).toJSON();

      expect(app).toMatchSnapshot();
    });
  });
});
