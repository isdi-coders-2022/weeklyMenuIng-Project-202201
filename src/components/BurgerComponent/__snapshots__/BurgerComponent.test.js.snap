// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a BurgerComponent component When it's rendered passing a function then it should match the snapshot 1`] = `
<div
  className="sc-bdvvtL hJvUau burger-container"
  title="Navigation Menu"
>
  <span
    className="sc-gsDKAQ bFRFTX burger-bar"
  />
  <span
    className="sc-gsDKAQ bFRFTX burger-bar"
  />
  <span
    className="sc-gsDKAQ bFRFTX burger-bar"
  />
</div>
`;
